host/phm-cli/src/cli.rs

```diff
@@ -0,0 +1,93 @@
+use std::num::ParseIntError;
+
+use clap::{Args, Parser, Subcommand};
+use embedded_hal::prelude::{_embedded_hal_blocking_i2c_Read, _embedded_hal_blocking_i2c_Write};
+use phm::Machine;
+use serialport::SerialPortInfo;
+
+#[derive(Parser, Debug)]
+pub enum PhmCli {
+    I2C(I2C),
+}
+
+#[derive(Parser, Debug)]
+pub struct I2C {
+    #[clap(subcommand)]
+    command: I2CCommand,
+}
+
+#[derive(Subcommand, Debug)]
+enum I2CCommand {
+    #[clap(name = "write")]
+    I2CWrite(I2CWrite),
+    #[clap(name = "read")]
+    I2CRead(I2CRead),
+    #[clap(name = "read-write")]
+    I2CReadWrite(I2CReadWrite),
+}
+
+#[derive(Args, Debug)]
+struct I2CWrite {
+    #[clap(short = 'a')]
+    address: u8,
+    #[clap(short = 'b', long = "write")]
+    write_bytes: String,
+}
+
+#[derive(Args, Debug)]
+struct I2CRead {
+    #[clap(short = 'a')]
+    address: u8,
+    #[clap(long = "read-ct")]
+    read_count: usize,
+}
+
+#[derive(Args, Debug)]
+struct I2CReadWrite {
+    #[clap(short = 'a')]
+    address: u8,
+    #[clap(short = 'b', long = "bytes")]
+    write_bytes: String,
+    #[clap(long = "read-ct")]
+    read_count: usize,
+}
+
+impl PhmCli {
+    pub fn run(machine: &mut Machine) -> Result<(), phm::Error> {
+        let cmd = PhmCli::parse();
```

Comment:
You might want to do the `::parse()` *before* connecting to the `Machine`, otherwise you won't be able to see `help` commands without having a machine connected